File: ecos-records/src/main/java/ru/citeck/ecos/records3/rest/v1/mutate/MutateResp.java
package ru.citeck.ecos.records3.rest.v1.mutate;

import lombok.Getter;
import lombok.Setter;
import ru.citeck.ecos.records3.record.op.atts.dto.RecordAtts;
import ru.citeck.ecos.records3.rest.v1.RequestResp;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class MutateResp extends RequestResp {

    @Getter
    private List<RecordAtts> records = new ArrayList<>();

    public void setRecords(List<RecordAtts> records) {
        if (records != null) {
            this.records = new ArrayList<>(records);
        } else {
            this.records.clear();
        }
    }

    public void addRecord(RecordAtts meta) {
        this.records.add(meta);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/rest/v1/mutate/MutateResp.kt
package ru.citeck.ecos.records3.rest.v1.mutate

import lombok.Getter
import lombok.Setter
import ru.citeck.ecos.records3.record.atts.dto.RecordAtts
import ru.citeck.ecos.records3.rest.v1.RequestResp
import java.util.*

@Getter
@Setter
class MutateResp : RequestResp() {

    var records: MutableList<RecordAtts> = ArrayList()
        private set

    fun setRecords(records: List<RecordAtts>?) {
        if (records != null) {
            this.records = ArrayList(records)
        } else {
            this.records = ArrayList()
        }
    }

    fun addRecord(meta: RecordAtts) {
        records.add(meta)
    }
}
